lib/cli.ts

```diff
@@ -84,6 +175,17 @@ if (argv.version) {
   process.exit(0);
 }
 
+if (!argv.disableChecks) {
+  // Check to see if additional flags were used.
+  let unknownKeys: string[] = Object.keys(argv).filter((element: string) => {
```

Comment:
(I'm not gonna make you do anymore with this, but it just occurred to me that [`Array.prototype.some`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some) is designed exactly for this situation:

```ts
let hasUnknownKeys = Object.keys(argv).some((element: string) => {
  return element !== '$0' && element !== '_' && allowedNames.indexOf(element) === -1;
});

if (hasUnknownKeys) {
  throw new Error('Found extra flags: ' + unknownKeys.join(', '));
}
```
It's so beautiful)